scripts/manage_case

```diff
@@ -1,375 +1,172 @@
-
-    my @files = $xml_config->findnodes(".//entry[\@id=\"MACHINES_SPEC_FILE\"]/default_value");
-    if ($#files == -1) {
-	die " ERROR: in parsing $file_config for the MACHIENS_SPEC_FILE node \n";
-    } 
-    my $file = $files[0]->textContent();
-    $file =~ s/\$CIMEROOT/$cimeroot/;
-    $file =~ s/\$MODEL/$model/;
-    (-f "$file")  or  die "** Cannot find machines specification file $file ***\n";
-
-    my $model = uc $model;
-    print "\n";
-    print " ======================================================\n";
-    print "            $model Machines:                          \n";
-    print " ======================================================\n";
-    print "\n";
-
-    my $xml_machine = XML::LibXML->new( no_blanks => 1)->parse_file($file);
-    foreach my $machine ($xml_machine->findnodes(".//machine")) {
-	my $name = $machine->getAttribute('MACH');
-	foreach my $node ($machine->findnodes("./DESC")) {
-	    my $desc = $node->textContent();
-	    print " $name :  \n      $desc \n";
-	}
-    }
-}
+#!/usr/bin/env python2.7
+#FIXME - the above needs the explicit python2.7 - this is not general ???
+
+from Tools.standard_script_setup import *
+
+from CIME.utils         import expect, get_model
+from CIME.XML.files     import Files
+from CIME.XML.component import Component
+from CIME.XML.compsets  import Compsets
+from CIME.XML.grids     import Grids
+from CIME.XML.machines  import Machines
+
+import argparse, doctest
+
+###############################################################################
+def query_grids():  
+###############################################################################
+
+    files = Files()
+    config_file = files.get_value("GRIDS_SPEC_FILE", resolved=True)
+    expect(os.path.isfile(config_file),
+           "Cannot find config_file %s on disk" %config_file)
+
+    parsed_xml = Grids(config_file)
+    if verbose:
+        parsed_xml.print_values(verbose=verbose)
+    else:
+        parsed_xml.print_values()
+
+###############################################################################
+def query_machines():  
+###############################################################################
+
+    files = Files()
+    config_file = files.get_value("MACHINES_SPEC_FILE", resolved=True)
+    expect(os.path.isfile(config_file),
+           "Cannot find config_file %s on disk" %config_file)
+
+    parsed_xml = Machines(config_file)
+    if verbose:
+        parsed_xml.print_values(verbose=verbose)
+    else:
+        parsed_xml.print_values()
+
+###############################################################################
+def query_compsets(name):  
+###############################################################################
+
+    # Determine valid component values by checking the value attributes for COMPSETS_SPEC_FILE
+    files = Files()
+    components = files.get_components("COMPSETS_SPEC_FILE", resolved=True)
+    match_found = "unset"
+    for component in components:
```

Comment:
I think that this can be reduced to 
  if name in components:
       match_found = name